### PR TITLE
Extended CSAF discovery

### PR DESCRIPTION
**Pull request by oxisto**
* Added new ontology classes for signatures and integrity
* Added discovery for CSAF advisory hashes
* Implementation of signature discovery

### CODE REVIEW

File: service/discovery/extra/csaf/document_properties.go [hunk 1]
```diff
@@ -70,3 +83,144 @@ func clientAuthenticity(res *http.Response) *ontology.Authenticity {
...
+// checksums located at the URL specified in the [csaf.AdvisoryFile] interface.
+//
+// It uses the [documentChecksum] method to retrieve the respective checksum.
+func (d *csafDiscovery) documentChecksums(file csaf.AdvisoryFile, body []byte) (checksums []*ontology.DocumentChecksum) {
+	var toHash map[string]hash.Hash = map[string]hash.Hash{
+		constants.SHA_256: sha256.New(),
+		constants.SHA_512: sha512.New(),
+	}
+
+	for algorithm, hash := range toHash {
```
Comment (lebogg) [inline service/discovery/extra/csaf/document_properties.go:98]: don't use package name 'hash' if possible, e.g. "h"

File: service/discovery/extra/csaf/provider_discover.go [hunk 2]
```diff
@@ -62,21 +64,42 @@ func (d *csafDiscovery) handleProvider(lpmd *csaf.LoadedProviderMetadata) (provi
 			SchemaUrl: "https://docs.oasis-open.org/csaf/csaf/v2.0/provider_json_schema.json",
 			Errors:    providerValidationErrors(lpmd.Messages),
 		},
+		Raw: discovery.Raw(pmd),
+	}
+
+	keys := d.discoverKeys(pmd.PGPKeys)
+	providers = append(providers, keys...)
```
Comment (lebogg) [inline service/discovery/extra/csaf/provider_discover.go:71]: Maybe append keys in the end of the function where the rest is appended as well. Meanwhile, we could add a comment to `handleProvider` including what ia actually returned (securiyAdvisoryDocuments, keys, ...)

**Review by lebogg — COMMENTED** [review]
LGTM, but in general more comments would be nice

File: api/ontology/ontology.proto [hunk 3]
```diff
@@ -1779,12 +1804,13 @@ message SecurityAdvisoryService {
   Authenticity authenticity = 9;
   optional string compute_id = 10;
   GeoLocation geo_location = 11;
-  repeated Redundancy redundancies = 12;
-  optional string parent_id = 13;
-  repeated string security_advisory_document_ids = 14;
-  optional string service_metadata_document_id = 15;
-  TransportEncryption transport_encryption = 16;
-  UsageStatistics usage_statistics = 17;
+  repeated string key_ids = 12;
```
[the remark this answers is not in the document]
Comment (oxisto) [reply api/ontology/ontology.proto:1807]: I can add it to the ontology, but it will not completely transfer to the protobuf yet.

**Review by lebogg — APPROVED** [review]

**Pull request merged**